test-utils/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+
+pub enum TestAction {
+    Add(String),
+    Remove(String),
+    Metadata,
+}
+
+/// Convert a vector of actions into a newline delimited json string
+pub fn actions_to_string(actions: Vec<TestAction>) -> String {
+    actions
+            .into_iter()
+            .map(|test_action| match test_action {
+                TestAction::Add(path) => format!(r#"{{"add":{{"path":"{path}","partitionValues":{{}},"size":262,"modificationTime":1587968586000,"dataChange":true, "stats":"{{\"numRecords\":2,\"nullCount\":{{\"id\":0}},\"minValues\":{{\"id\": 1}},\"maxValues\":{{\"id\":3}}}}"}}}}"#),
+                TestAction::Remove(path) => format!(r#"{{"remove":{{"path":"{path}","partitionValues":{{}},"size":262,"modificationTime":1587968586000,"dataChange":true}}}}"#),
+                TestAction::Metadata => METADATA.into(),
+            })
+            .join("\n")
+}
+
+/// convert a RecordBatch into a vector of bytes. We can't use `From` since these are both foreign
+/// types
+pub fn record_batch_to_bytes(batch: &RecordBatch) -> Vec<u8> {
+    let mut data: Vec<u8> = Vec::new();
+    let props = WriterProperties::builder().build();
+    let mut writer = ArrowWriter::try_new(&mut data, batch.schema(), Some(props)).unwrap();
+    writer.write(batch).expect("Writing batch");
+    // writer must be closed to write footer
+    writer.close().unwrap();
+    data
+}
+
+/// Anything that implements `IntoArray` can turn itself into a reference to an arrow array
+pub trait IntoArray {
+    fn into_array(self) -> ArrayRef;
+}
+
+impl IntoArray for Vec<i32> {
+    fn into_array(self) -> ArrayRef {
+        Arc::new(Int32Array::from(self))
+    }
+}
+
+impl IntoArray for Vec<&'static str> {
+    fn into_array(self) -> ArrayRef {
+        Arc::new(StringArray::from(self))
+    }
+}
+
+/// Generate a record batch from an iterator over (name, array) pairs. Each pair specifies a column
+/// name and the array to associate with it
+pub fn generate_batch<I, F>(items: I) -> Result<RecordBatch, ArrowError>
+where
+    I: IntoIterator<Item = (F, ArrayRef)>,
+    F: AsRef<str>,
+{
+    RecordBatch::try_from_iter(items)
+}
+
+/// Generate a RecordBatch with two columns (id: int, val: str), with values "1,2,3" and "a,b,c"
+/// respectively
+pub fn generate_simple_batch() -> Result<RecordBatch, ArrowError> {
+    generate_batch(vec![
+        ("id", vec![1, 2, 3].into_array()),
+        ("val", vec!["a", "b", "c"].into_array()),
+    ])
+}
+
+/// put a commit file into the specified object store.
+pub async fn add_commit(
+    store: &dyn ObjectStore,
+    version: u64,
+    data: String,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let filename = format!("_delta_log/{:0>20}.json", version);
```

Comment:
I've seen this line of code a few times in the codebase. Might be worth making it into a common util `delta_path_from_version` or smth.

Reply:
> I've seen this line of code a few times in the codebase. Might be worth making it into a common util `delta_path_from_version` or smth.

Good call, done